pyp/pyp.py

```diff
@@ -20,22 +24,24 @@ def pip_freeze(package_names):
     ).decode('ascii')
     return output
 
+def get_existing_entries():
+    entries = []
+    try:    
+        entries = [entry.rstrip('\n') for entry in open(REQUIREMENTS_FILE_NAME)]
+    except FileNotFoundError:
+        open(REQUIREMENTS_FILE_NAME, 'w').close()
+    
+    return entries
 
 def write_requirements(packages):
-    for pkg in packages.split('\n')[:-1]:  # Skip the '' at the end
-        while True:
-            try:
-                with open('requirements.txt', 'r+') as f:
-                    for line in f:
-                        if pkg in line:
-                            break
-                    else:
-                        f.write(pkg + '\n')
-            except FileNotFoundError:
-                open('requirements.txt', 'w').close()
-            else:
-                break
-
+    existing_entries = get_existing_entries()
+    new_entries = packages.split('\n')[:-1] # Skip the '' at the end
+    combined_entries = existing_entries + new_entries
```

Comment:
I noticed a small bug with this line. If you `pyp install` a package twice, the package will get written twice to the `requirements.txt`. Ideally, `pyp install` should be idempotent. I think an efficient way to fix this would be to import `OrderedDict` from `collections` and do
```suggestion
    combined_entries = list(OrderedDict.fromkeys(existing_entries + new_entries))
```
The reason I suggest using `OrderedDict` is because it removes duplicates while preserving the order of `existing_entries` which means the sorting algorithm doesn't have to do as much.

Reply:
Hey, that's a very nice catch! The approach suggested was a bit of a mind-bender for me, but I eventually managed to understand it. 😅

Reply:
Glad you figured it out!